### PR TITLE
Adding downward API for the namespace of the syncer iamge

### LINKED ISSUE #83: Don't sync anything in the syncer-system namespace
The in-cluster syncer runs in `syncer-system`, and today if you create something in that namespace in kcp, the syncer will apply it to its own local cluster, possibly even replacing/breaking its own deployment.

Let's just keep syncer from messing with its own namespace entirely, to avoid a whole bunch of problems.

### PR DESCRIPTION
**Pull request by shawn-hurley**
* Removing processing of items in the same namespace as the syncer.

fixes #83

### CODE REVIEW

**Comment by shawn-hurley** [comment]
@imjasonh @davidfestal 

I think that this may be one solution. I think the other solution is to get the object meta and check the namespace in the `eventhandler`, this would I think keep workers from getting items but I didn't know if this was something that was desired, as I was wondering if the tombstone check could also be there?

Let me know if you would like this approach and I can change it for that.

File: pkg/syncer/syncer.go [hunk 1]
```diff
@@ -23,6 +24,7 @@ import (
 )
 
 const resyncPeriod = 10 * time.Hour
+const SYNCER_NAMESPACE_KEY = "SYNCER_NAMESPACE"
```
Comment (imjasonh) [inline pkg/syncer/syncer.go:27]: style nit: typically this would just be `SyncerNamespaceKey`

File: pkg/syncer/syncer.go [hunk 2]
```diff
@@ -91,8 +96,9 @@ func New(from, to *rest.Config, upsertFn UpsertFunc, deleteFn DeleteFunc, handle
 
 		stopCh: stopCh,
 
-		upsertFn: upsertFn,
-		deleteFn: deleteFn,
+		upsertFn:  upsertFn,
+		deleteFn:  deleteFn,
+		namespace: os.Getenv(SYNCER_NAMESPACE_KEY),
```
Comment (imjasonh) [inline pkg/syncer/syncer.go:101]: How should syncer react when this env var is unset? Should it panic, or fallback to some default?

File: pkg/syncer/syncer.go [hunk 3]
```diff
@@ -337,3 +347,16 @@ func (c *Controller) getClient(gvr schema.GroupVersionResource, namespace string
 	}
 	return nri
 }
+
+func (c *Controller) inSyncerNamesapce(objectNamespace string) bool {
```
Comment (imjasonh) [inline pkg/syncer/syncer.go:351]: typo: `inSyncerNamespace`

**Review by imjasonh — COMMENTED** [review]
This approach seems fine to me, at least for now. Other options definitely exist, but none stick out to me as noticeably better. Thanks for taking this on!

File: pkg/syncer/syncer.go (hunk 2 above)
Comment (shawn-hurley) [reply pkg/syncer/syncer.go:101]: that should be I thought covered in `inSyncerNamespace`

**Review by davidfestal — APPROVED** [review]

**Review by imjasonh — APPROVED** [review]

**Pull request merged**